spec/02-integration/05-proxy/05-ssl_spec.lua

```diff
@@ -42,7 +42,30 @@ describe("SSL", function()
       http_if_terminated = false,
     })
 
-    assert(helpers.start_kong())
+    assert(helpers.dao.apis:insert {
+      name = "api-3",
+      hosts = { "ssl3.com" },
+      upstream_url = "https://localhost:10001",
+      preserve_host = true,
+    })
+
+    assert(helpers.dao.apis:insert {
+      name = "api-4",
+      hosts = { "no-sni.com" },
+      upstream_url = "https://localhost:10001",
+      preserve_host = false,
+    })
+
+    assert(helpers.dao.apis:insert {
+      name = "api-5",
+      hosts = { "nil-sni.com" },
+      upstream_url = "https://127.0.0.1:10001",
+      preserve_host = false,
+    })
+
+    assert(helpers.start_kong {
+      nginx_conf = "spec/fixtures/custom_nginx.template",
```

Comment:
As per our discussion on this test, we should include a strongly opinionated comment that such usage of the custom Nginx template is only made because it is our only viable solution for this test as of this day.